### PR TITLE
Chapter1/homework/5

### PR DESCRIPTION
**Pull request by Smephey**
Submitting solution for chapter1/homework/5.

### CODE REVIEW

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 1]
```diff
@@ -2,94 +2,166 @@
...
     {
-        /// <summary>
-        /// Sort the array in ascending order.
-        /// If array empty or null- don't do anything.
-        /// </summary>
-        /// <param name="array">Input array in a random order.</param>
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs:7]: Try inverting the if statements and return. This reduces the nested scopes number.
```
if(array == null || array.Legnth == 0) return;
int t; //...
```

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 2]
```diff
@@ -2,94 +2,166 @@
...
+
+            }
         }
 
-        /// <summary>
-        /// Remove last element in array.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <returns>A new array with the last element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveLast(int[] array)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs:40]: Reuse ``RemoveAt(arrau. lastIndex)``

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 3]
```diff
@@ -2,94 +2,166 @@
...
+            }
+
             return array;
         }
 
-        /// <summary>
-        /// Remove first element in array.
-        /// </summary>
-        /// <returns>A new array with the first element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveFirst(int[] array)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs:55]: Reuse RemoveAt

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 4]
```diff
@@ -2,94 +2,166 @@
...
             return array;
         }
 
-        /// <summary>
-        /// Inserts a new array element at the start.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs:94]: Reuse InsertAt

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 5]
```diff
@@ -2,94 +2,166 @@
...
         }
 
-        /// <summary>
-        /// Inserts a new array element at the end.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <returns>A new array with element added in the end of array. If an array is empty or null, returns new array with number in it.</returns>
+
         public static int[] InsertLast(int[] array, int number)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs:114]: Reuse InsertAt

**Comment by Smephey** [comment]
> Please be more mindful to the function that you can reuse in other functions.

Oh of course, that's so obvious! Can't believe I didn't consider that.

**Comment by Smephey** [comment]
Thanks!

**Pull request closed without merging**